proto/task_execution.proto

```diff
@@ -0,0 +1,132 @@
+
+
+syntax = "proto3";
+
+package ga4gh_task_exec;
+
+// Import HTTP RESTful annotations
+import "google/api/annotations.proto";
+
+/// Mapping long term file storage to path used in docker container
+message FileParameter {
+  string path = 1;        /// Path inside the docker container
+  string storagePath = 2; /// Path of the data's long term storage
+}
+
+/// A command line to be executed and the docker container to run it
+message DockerExecutor {
+  string imageName = 1;   /// Docker Image name
+  string cmd = 2;         /// The command to be executed
+}
+
+/// Attached disk request. 
+message Disk {
+  string name = 1;        /// Name of attached disk
+  uint32 sizeGb = 2;      /// Minium size
+  string source = 3;      /// Source data, this would refer to an existing disk the execution engine
+                          /// could identify. Leave blank if is to be a newly created disk
+  bool autoDelete = 4;    /// Automatically delete after usage
+  bool readOnly = 5;      /// mount into docker as read only
+  string mountPoint = 6;  /// mount point for disk inside the docker container
+}
+
+message Resources {
+  uint32 minimumCpuCores = 1;
+  bool preemptible = 2;
+  uint32 minimumRamGb = 3;
+  repeated Disk disks = 4;
```

Comment:
@ccarrizo Do you have suggestions on the concrete format?

I like the idea on paper but IMO something like this needs to be simple either via a) fairly limited syntax or b) very generic syntax which allows for arbitrary constructs

I think 'a' is better here so I'd want to see what the extra syntax would look like.